smtp_connector.py

```diff
@@ -63,106 +157,289 @@ def _validate_email(self, input_data):
-        """
-        This method returns the encoded|original string based on the Python version.
-        :param input_str: Input string to be processed
-        :return: input_str (Processed input string based on following logic 'input_str - Python 3; encoded input_str - Python 2')
-        """
+        ret_val, resp_json = self.make_rest_call(action_result, '{}rest/system_info'.format(self.get_phantom_base_url()))
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), None
+
+        phantom_base_url = resp_json.get('base_url')
+        if not phantom_base_url:
+            return action_result.set_status(
+                phantom.APP_ERROR, "SOAR Base URL is not configured, please configure it in System Settings"), None
+
+        phantom_base_url = phantom_base_url.strip("/")
+
+        return phantom.APP_SUCCESS, phantom_base_url
+
+    def _get_asset_name(self, action_result):
+
+        ret_val, resp_json = self.make_rest_call(
+            action_result, '{}rest/asset/{}'.format(self.get_phantom_base_url(), self.get_asset_id()))
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), None
+
+        asset_name = resp_json.get('name')
+        if not asset_name:
+            return action_result.set_status(phantom.APP_ERROR, "Error retrieving asset name"), None
+
+        return phantom.APP_SUCCESS, asset_name
+
+    def _get_url_to_app_rest(self, action_result=None):
+        if not action_result:
+            action_result = ActionResult()
+        # get the phantom ip to redirect to
+        ret_val, phantom_base_url = self._get_phantom_base_url_smtp(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), action_result.get_message()
+        # get the asset name
+        ret_val, asset_name = self._get_asset_name(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), action_result.get_message()
+        self.save_progress('Using SOAR base URL as: {0}'.format(phantom_base_url))
+        app_json = self.get_app_json()
+        app_name = app_json['name']
+        app_dir_name = _get_dir_name_from_app_name(app_name)
+        url_to_app_rest = "{0}/rest/handler/{1}_{2}/{3}".format(phantom_base_url, app_dir_name, app_json['appid'], asset_name)
+        return phantom.APP_SUCCESS, url_to_app_rest
+
+    def _interactive_auth_initial(self, client_id, rsh, client_secret):
+
+        state = rsh.load_state()
+        asset_id = self.get_asset_id()
+
+        ret_val, app_rest_url = self._get_url_to_app_rest()
+        if phantom.is_fail(ret_val):
+            return phantom.APP_ERROR, app_rest_url
+
+        config = self.get_config()
+
+        request_url = config.get("auth_url", "").strip("\\/")
+        token_url = config.get("token_url", "").strip('\\/')
+        # set proxy if configured
+        proxy = {}
+        if 'HTTP_PROXY' in os.environ:
```

Comment:
Check proxy here too, if you haven't already

Reply:
To pass the values to the request handler we have added the proxies here explicitly. We have tested it with no-proxy, http-proxy and https-proxy.